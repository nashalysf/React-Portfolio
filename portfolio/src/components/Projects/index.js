import React, { useState } from 'react';
import Modal from '../Modal';

const Projects = ({ category }) => {
  const [isModalOpen, setIsModalOpen] = useState(false);
  const [currentPhoto, setCurrentPhoto] = useState();
  const [photos] = useState ([
    {
      name: 'Horiseon',
      category: 'front',
      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie',
    },
    {
      name: 'Find-A-Story',
      category: 'front',
      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie',
    },
    {
      name: 'Password-Generator',
      category: 'front',
      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie',
    },
    {
      name: 'Story-Keeper',
      category: 'back',
      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie',
    },
    {
      name: 'Deepen',
      category: 'back',
      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie',
    },
    {
      name: 'Just-Tech-news',
      category: 'back',
      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie',
    },
  ]);
  const currentPhotos = photos.filter((photo)=>photo.category === category);
  const toggleModal = (image, i) => {
    setCurrentPhoto({ ...image, index: i });
    setIsModalOpen(!isModalOpen);
  }
  return (
    <div>
    {isModalOpen && (
      <Modal onClose={toggleModal} currentPhoto={currentPhoto} />
    )}
    <div className="flex-row">
      {currentPhotos.map((image, i) => (
        <img
          src={require(`../../assets/small/${category}/${i}.jpg`).default}
          alt={image.name}
          className="img-thumbnail mx-1"
          onClick={() => toggleModal(image, i)}
          key={image.name}
        />
      ))}
    </div>
  </div>
  )
}
  
export default Projects;
  

