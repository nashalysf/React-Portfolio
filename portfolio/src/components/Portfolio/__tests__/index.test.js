// __tests__/Gallery.test.js
import React from 'react'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Projects from '..'
const back = { name: "Deepen", description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie" };

afterEach(cleanup)

describe('Portfolio component', () => {

  it('renders', () => {
    render(<Projects currentCategory={back} />);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Projects currentCategory={back} />)
    expect(asFragment()).toMatchSnapshot()
  })
});

it('renders', () => {
  const { getByTestId } = render(<Projects currentCategory={back} />)
  // eslint-disable-next-line testing-library/prefer-screen-queries
  expect(getByTestId('h1tag')).toHaveTextContent('Deepen')
})