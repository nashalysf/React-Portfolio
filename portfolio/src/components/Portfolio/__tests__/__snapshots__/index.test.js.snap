// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio component matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <h1
      data-testid="h1tag"
    >
      Deepen
    </h1>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie
    </p>
    <div>
      <div
        class="flex-row"
      />
    </div>
  </section>
</DocumentFragment>
`;
